File: 6sem(curr)/Mobile_Aps/lab01/task_2/in_java/src/main/java/com/example/l1z2_java/MainActivity.java
package com.example.l1z2_java;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import java.util.Random;

public class MainActivity extends AppCompatActivity {
    private int number = 0;
    private int guesses = 0;
    private boolean gameON = true;
    private final Random g = new Random();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        setup();
    }

    private void setup() {
        this.guesses = 0;
        this.gameON = true;
        this.number = g.nextInt(1000);
        TextView guesses = (TextView) findViewById(R.id.gesses_counter);
        TextView gameInfo = (TextView) findViewById(R.id.game_info);
        EditText input = (EditText) findViewById(R.id.editTextNumber);
        guesses.setText("Guesses: 0");
        gameInfo.setText("Guess the random number. \nThe number is in range: \nfrom 1 to 1000.");
        input.getText().clear();
    }

    public void checkInput(View view) {
        if (gameON) {
            EditText input = (EditText) findViewById(R.id.editTextNumber);
                    int guess;
            try {
                guess = Integer.parseInt(input.getText().toString());
                input.getText().clear();
            } catch (NumberFormatException ne) {
                Toast.makeText(this, "Wrong input!", Toast.LENGTH_SHORT).show();
                input.getText().clear();
                return;
            }
            guesses++;
            TextView guesses = (TextView) findViewById(R.id.gesses_counter);
            guesses.setText("Guesses: " + this.guesses);

            TextView gameInfo = (TextView) findViewById(R.id.game_info);
            if (guess == number) {
                gameInfo.setText("Congratulations!!! \nYou guessed correctly.");
                gameON = false;
            }

            else if (guess < number) {
                gameInfo.setText("Wrong! \nThe number is higher.");
            }
            else {
                gameInfo.setText("Wrong! \nThe number is lower.");
            }
        }
    }

    public void restartGame(View view) {
        setup();
    }
}

File: 6sem(curr)/Mobile_Aps/lab01/task_2/in_kotlin/src/main/java/com/example/l1z2_kotlin/MainActivity.kt
package com.example.l1z2_kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import kotlin.random.Random


class MainActivity : AppCompatActivity() {
    private var number = 0
    private var guesses = 0
    private var gameON = true


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setup()
    }

    private fun setup() {
        guesses = 0
        gameON = true
        number = Random.nextInt(1000)
        findViewById<TextView>(R.id.gesses_counter).text = "Guesses: 0"
        findViewById<TextView>(R.id.game_info).text = "Guess the random number. \nThe number is in range: \nfrom 1 to 1000."
        findViewById<EditText>(R.id.editTextNumber).text.clear()
    }

    fun checkInput(view: View) {
        if (gameON) {
            val input = findViewById<EditText>(R.id.editTextNumber)
            val guess: Int
            try {
                    guess = input.text.toString().toInt()
                    input.text.clear()
            } catch (e: java.lang.NumberFormatException) {
                Toast.makeText(this, "Wrong input!", Toast.LENGTH_SHORT).show()
                input.text.clear()
                return
            }
            guesses++
            findViewById<TextView>(R.id.gesses_counter).text = "Guesses: $guesses"

            if (guess == number) {
                findViewById<TextView>(R.id.game_info).text = "Congratulations!!! \nYou guessed correctly."
                gameON = false
            }
            else if (guess < number) {
                findViewById<TextView>(R.id.game_info).text = "Wrong! \nThe number is higher."
            }
            else {
                findViewById<TextView>(R.id.game_info).text = "Wrong! \nThe number is lower."
            }
        }
    }

    fun restartGame(view: View) {
        setup()
    }
}
